homework/index.js

```diff
@@ -27,21 +31,145 @@
         elem.setAttribute(key, value);
       }
     });
+
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-      } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
-      }
+  function addRow(parent, labelText, rowValue) {
+    const row = createAndAppend('tr', parent);
+    createAndAppend('td', row, {
+      class: 'label',
+      text: labelText,
+    });
+    createAndAppend('td', row, {
+      text: rowValue,
+    });
+    return row;
+  }
+
+  function renderRepositoryInfo(repository, leftContainer) {
+    const table = createAndAppend('table', leftContainer);
+    const tbody = createAndAppend('tbody', table);
+    const firstRow = addRow(tbody, 'Repository : ', '');
+    createAndAppend('a', firstRow.children[1], {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
+    });
+    addRow(tbody, 'Description :', repository.description);
+    addRow(tbody, 'Forks :', repository.forks);
+    addRow(tbody, 'Updated :', repository.updated_at);
+  }
+
+  function renderRepoContributors(contributors, rightContainer) {
+    createAndAppend('p', rightContainer, {
+      class: 'contributor-header',
+      text: 'Contributors',
+    });
+    const ul = createAndAppend('ul', rightContainer, {
+      class: 'contributor-list',
+    });
+    contributors.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-container',
+      });
+      createAndAppend('img', li, {
+        class: 'contributor-avatar',
+        src: contributor.avatar_url,
+      });
+      const contributorDataDiv = createAndAppend('div', li, {
+        class: 'contributor-data',
+      });
+      createAndAppend('a', contributorDataDiv, {
+        text: contributor.login,
+        href: contributor.html_url,
+        class: 'contributor-name',
+      });
+      createAndAppend('div', contributorDataDiv, {
+        text: contributor.contributions,
+        class: 'contributionCount',
+      });
+    });
+  }
+
+  function fetchContributors(url, rightContainer) {
+    const fetchedContributors = fetchJSON(url);
+    fetchedContributors
+      .then(contributors => {
+        renderRepoContributors(contributors, rightContainer);
+        return fetchedContributors;
```

Comment:
You do not need to return anything here. What you return from a `.then()` method is passed as a parameter to the next `.then()` in the chain. But there is no next `.then()`, only a `catch().`